core/blockchain.go

```diff
@@ -775,31 +775,37 @@ func (self *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 }
 
 func (self *BlockChain) update() {
-	events := self.eventMux.Subscribe(queueEvent{})
+	// Subscribe to blockchain insertion event sets
+	eventCh := self.eventMux.Subscribe(queueEvent{}).Chan()
```

Comment:
We could rework that part (also an inoffensive bug I've noticed, is that after chain insertions, a lot of nil events are contained in the queue (ignored blocks), which get discarded, but are still passed through the entire event system). I'd propose leaving a bigger refactor for later so that we could merge this into develop and fix the Wallet "freezing" issue.